runtime/api/kotlinx-benchmark-runtime.klib.api

```diff
@@ -378,78 +99,8 @@ final class kotlinx.benchmark/Blackhole { // kotlinx.benchmark/Blackhole|null[0]
-        final fun <set-f1>(kotlin/Float) // kotlinx.benchmark/Blackhole.f1.<set-f1>|<set-f1>(kotlin.Float){}[0]
-    final var i0 // kotlinx.benchmark/Blackhole.i0|{}i0[0]
-        final fun <get-i0>(): kotlin/Int // kotlinx.benchmark/Blackhole.i0.<get-i0>|<get-i0>(){}[0]
-        final fun <set-i0>(kotlin/Int) // kotlinx.benchmark/Blackhole.i0.<set-i0>|<set-i0>(kotlin.Int){}[0]
-    final var i1 // kotlinx.benchmark/Blackhole.i1|{}i1[0]
-        final fun <get-i1>(): kotlin/Int // kotlinx.benchmark/Blackhole.i1.<get-i1>|<get-i1>(){}[0]
-        final fun <set-i1>(kotlin/Int) // kotlinx.benchmark/Blackhole.i1.<set-i1>|<set-i1>(kotlin.Int){}[0]
-    final var l0 // kotlinx.benchmark/Blackhole.l0|{}l0[0]
-        final fun <get-l0>(): kotlin/Long // kotlinx.benchmark/Blackhole.l0.<get-l0>|<get-l0>(){}[0]
-        final fun <set-l0>(kotlin/Long) // kotlinx.benchmark/Blackhole.l0.<set-l0>|<set-l0>(kotlin.Long){}[0]
-    final var l1 // kotlinx.benchmark/Blackhole.l1|{}l1[0]
-        final fun <get-l1>(): kotlin/Long // kotlinx.benchmark/Blackhole.l1.<get-l1>|<get-l1>(){}[0]
-        final fun <set-l1>(kotlin/Long) // kotlinx.benchmark/Blackhole.l1.<set-l1>|<set-l1>(kotlin.Long){}[0]
-}
-// Targets: [native]
-final class kotlinx.benchmark/NativeIntelliJBenchmarkProgress : kotlinx.benchmark/IntelliJBenchmarkProgress { // kotlinx.benchmark/NativeIntelliJBenchmarkProgress|null[0]
-    constructor <init>(kotlin/String) // kotlinx.benchmark/NativeIntelliJBenchmarkProgress.<init>|<init>(kotlin.String){}[0]
-    final fun endBenchmarkException(kotlin/String, kotlin/String, kotlin/String, kotlin/String) // kotlinx.benchmark/NativeIntelliJBenchmarkProgress.endBenchmarkException|endBenchmarkException(kotlin.String;kotlin.String;kotlin.String;kotlin.String){}[0]
-    final fun startBenchmark(kotlin/String, kotlin/String) // kotlinx.benchmark/NativeIntelliJBenchmarkProgress.startBenchmark|startBenchmark(kotlin.String;kotlin.String){}[0]
-}
-// Targets: [native]
-final enum class kotlinx.benchmark/NativeFork : kotlin/Enum<kotlinx.benchmark/NativeFork> { // kotlinx.benchmark/NativeFork|null[0]
-    enum entry PerBenchmark // kotlinx.benchmark/NativeFork.PerBenchmark|null[0]
-    enum entry PerIteration // kotlinx.benchmark/NativeFork.PerIteration|null[0]
-    final fun valueOf(kotlin/String): kotlinx.benchmark/NativeFork // kotlinx.benchmark/NativeFork.valueOf|valueOf#static(kotlin.String){}[0]
-    final fun values(): kotlin/Array<kotlinx.benchmark/NativeFork> // kotlinx.benchmark/NativeFork.values|values#static(){}[0]
-    final val entries // kotlinx.benchmark/NativeFork.entries|#static{}entries[0]
-        final fun <get-entries>(): kotlin.enums/EnumEntries<kotlinx.benchmark/NativeFork> // kotlinx.benchmark/NativeFork.entries.<get-entries>|<get-entries>#static(){}[0]
 }
 // Targets: [js.jsIr, wasmJs]
 final class kotlinx.benchmark/Blackhole : kotlinx.benchmark/CommonBlackhole { // kotlinx.benchmark/Blackhole|null[0]
```

Comment:
It's not directly related to the change, but as soon as `CommonBlackhole` is hidden, there will be no info about `Blackhole`'s members in the dump.

There's a request to overcome that problem on BCV side and dump all members inherited from the private API (https://github.com/Kotlin/binary-compatibility-validator/issues/217), but here, I believe, the better solution would be to introduce a source set descending from `commonMain` and shared between both `jsMain` and `wasmJsMain` source sets and actualize the `Blackhole` there (kotlinx-io use that approach for implementing nodeJs-dependent classes for js and wasmJs: https://github.com/Kotlin/kotlinx-io/blob/c2d52202984f610054f21fd77def42e6f554bb70/build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-multiplatform.gradle.kts#L84)

Reply:
Thanks for pointing out. It turns out `CommonBlackhole` can't be properly made `internal`. Because in Kotlin public classes can't inherit from internal classes. I have created a separate PR for sharing `Blackhole` implementation btw Js and WasmJs, so that `CommonBlackhole ` could be dropped later: https://github.com/Kotlin/kotlinx-benchmark/pull/207